### PR TITLE
Review corrections to Easy Image linking

### PR DESCRIPTION
**Pull request by mlewand**
## What is the purpose of this pull request?

Other

### This PR contains

- [x] Unit tests
- [x] Manual tests (manual tests were already in place)

## What changes did you make?

Set of review corrections to #1137.

Most of changes are just:

* Fix for features not to override already registered feature information.
* Renamed `widgetFeatures` to `features`.
* Additional test cases.
* Test simplifications.

### CODE REVIEW

File: tests/plugins/imagebase/features/link.js [hunk 1]
```diff
@@ -44,45 +44,32 @@
 		} );
 	}
 
-	function assertLinkWidget( options ) {
+	function assertLinkWidgetStatus( options, linkCount, linkCmdState, unlinkCmdState ) {
```
Comment (Comandeer) [inline tests/plugins/imagebase/features/link.js]: I don't have today time to do full review, but it's the first thing that caught my eye: why did you expand `options` parameter to 4 parameters?

Comment (mlewand) [reply tests/plugins/imagebase/features/link.js]: This is just the way diff has shown it 😉 `assertLinkWidget` function is still there, but few lines below. I simply extracted common part of both functions.

Comment (Comandeer) [reply tests/plugins/imagebase/features/link.js]: OK, but why you expanded `options` then? :P

Comment (mlewand) [reply tests/plugins/imagebase/features/link.js]: Actually it make sense to reuse options, true.

Comment (mlewand) [reply tests/plugins/imagebase/features/link.js]: Well, true, it would be a better idea to extend options for that. 🙂

**Pull request merged**